bw2data/backends/proxies.py

```diff
@@ -330,12 +331,14 @@ def save(self, signal = True):
 
         databases.set_dirty(self["database"])
 
-        check_activity_type(self._data.get("type"))
-        check_activity_keys(self)
+        if not data_already_set:
+            check_activity_type(self._data.get("type"))
+            check_activity_keys(self)
+
+            for key, value in dict_as_activitydataset(self._data).items():
+                if key != "id":
+                    setattr(self._document, key, value)
```

Comment:
These seem to be idempotent, so I'm not sure the `data_already_set` is worth the added complexity.

Reply:
The typo checks take some resources, but that's only relevant when applying many changes.

Going from `_data` to `_document` when we just went the other way is not great - in theory it should be idempotent but I worry about things breaking here for stupid reasons, so would prefer to avoid the possibility of bugs coming in here by just skipping it at this point.

Reply:
This is now necessary because we don't actually set `._data` during revision application, so we need to skip this code. It's not great, but neither is the underlying architecture, and making small changes won't fix the real problems, so let's just hold our nose and keep it for now 😛